service/credential/credential.go

```diff
@@ -64,6 +66,111 @@ func GetAzureConfig(k8sClient kubernetes.Interface, name string, namespace strin
 	return c, nil
 }
 
+func GetAzureConfigFromSecret(credential *v1.Secret) (*client.AzureClientSetConfig, error) {
```

Comment:
Nice! :)

Reply:
(I was referring to all changes in `credential.go`, nicely extracted functionality)